sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -272,6 +244,44 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
         Nil
     }
 
+  override protected[sql] def createSession(): SQLSession = {
+    new this.SQLSession()
+  }
+
+  protected[hive] class SQLSession extends super.SQLSession {
```

Comment:
I think there's no need to overwrite `SQLSession` and `createSession` here, for `SessionState` self is `ThreadLocal`. we just need to set `SessionState` when `openSession` in `SparkSQLSessionManager`.